src/ostree/ot-remote-builtin-add.c

```diff
@@ -66,6 +69,20 @@ ot_remote_builtin_add (int argc, char **argv, GCancellable *cancellable, GError
                                     OSTREE_BUILTIN_FLAG_NONE, &repo, cancellable, error))
     goto out;
 
+  /* As a special case, we can take a --sysroot argument. Currently we also
+   * require --repo because fixing that needs more cmdline rework.
+   */
```

Comment:
You could pass `OSTREE_BUILTIN_FLAG_NO_REPO` to `ostree_option_context_parse` and define your own `--repo` option here so it's not required in the presence of `--sysroot`.